### PR TITLE
Add Black/White cursor support

### PR DESCRIPTION
**Pull request by ihhub**
For now only through configuration file.

image

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @oleg-derevenetz , I think we still have issues with compilation for PS Vita.

**Comment by oleg-derevenetz** [comment]
>  I think we still have issues with compilation for PS Vita.

Hi @ihhub these errors seems legitimate - alignment requirements of `int16_t` are higher than alignment requirements of `char`, it may be dangerous to just use `reinterpret_cast` in this case.

**Comment by ihhub** [comment]
> > I think we still have issues with compilation for PS Vita.
> 
> Hi @ihhub these errors seems legitimate - alignment requirements of `int16_t` are higher than alignment requirements of `char`, it may be dangerous to just use `reinterpret_cast` in this case.

It's a fair point. I'll change the logic.

**Comment by oleg-derevenetz** [comment]
> > > I think we still have issues with compilation for PS Vita.
> > 
> > Hi @ihhub these errors seems legitimate - alignment requirements of `int16_t` are higher than alignment requirements of `char`, it may be dangerous to just use `reinterpret_cast` in this case.
> 
> It's a fair point. I'll change the logic.

@ihhub `serialize.h` has the `getLEValue()` for such cases to safely load some numeric value stored in the little-ending format from the data file.

**Comment by oleg-derevenetz** [comment]
However streams should do the job as well.

**Review by oleg-derevenetz — COMMENTED** [review]
Hi @ihhub we can call these cursors "monochrome cursor" or `MONO_CURSOR_XXX`. What do you think?

**Comment by ihhub** [comment]
> Hi @ihhub we can call these cursors "monochrome cursor" or `MONO_CURSOR_XXX`. What do you think?

Yes, it is a good idea.

File: src/fheroes2/agg/icn.h [hunk 1]
```diff
@@ -963,6 +964,10 @@ namespace ICN
         EVIL_ARMY_BUTTON,
         EVIL_MARKET_BUTTON,
 
+        MONO_CURSOR_ADVMBW,
+        MONO_CURSOR_SPELBW,
+        MONO_CURSOR_CMSSBW,
+
```
Comment (oleg-derevenetz) [inline src/fheroes2/agg/icn.h:970]: Just curious, what those BW mean at the end?

**Review by oleg-derevenetz — APPROVED** [review]

File: src/fheroes2/agg/icn.h (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/agg/icn.h:970]: Black-White :) It comes from the naming of the original resources.

**Pull request merged**